app/services/checksum_validator.rb

```diff
@@ -2,86 +2,81 @@
     # in that AuditResults instance.  so set AuditResults#actual_version after both instances have been created.
     @results.actual_version = moab.current_version_id
   end
 
   def validate_checksums
     # check first thing to make sure the moab is present on disk, otherwise weird errors later
-    return results.result_array unless check_file_existence!
+    return persist! { moab_validator.mark_moab_not_found } unless moab_exists?
 
+    # These will populate the results object
     validate_manifest_inventories
     validate_signature_catalog
 
-    # This is to deal with db connection timeouts.
-    ActiveRecord::Base.clear_active_connections!
-
-    transaction_ok = ActiveRecordUtils.with_transaction_and_rescue(results) do
+    persist!(clear_connections: true) do
       complete_moab.last_checksum_validation = Time.current
       if results.result_array.empty?
         results.add_result(AuditResults::MOAB_CHECKSUM_VALID)
-        found_expected_version = moab.current_version_id == complete_moab.version
-        set_status_as_seen_on_disk(found_expected_version)
         complete_moab.update_audit_timestamps(true, true)
-        unless found_expected_version
-          results.add_result(AuditResults::UNEXPECTED_VERSION,
-                             actual_version: moab.current_version_id,
-                             db_obj_name: 'CompleteMoab',
-                             db_obj_version: complete_moab.version)
-        end
+
+        validate_versions
       else
-        update_status('invalid_checksum')
+        update_complete_moab_status('invalid_checksum')
       end
-      complete_moab.save!
     end
-    results.remove_db_updated_results unless transaction_ok
+  end
 
-    results.report_results
+  # @return [Moab::StorageObjectVersion]
+  def latest_moab_version
+    @latest_moab_version ||= moab.version_list.last
   end
 
   private
 
   # return true and exit if the moab exists.  otherwise update status to missing on complete_moab, report results, and return false.
-  # @return [Boolean] true and exit if the moab exists, false otherwise
-  def check_file_existence!
-    return true if File.exist?(object_dir) && latest_moab_version.present?
+  # @return [Boolean] true if the moab exists, false otherwise
+  def moab_exists?
+    File.exist?(object_dir) && latest_moab_version.present?
+  end
 
-    transaction_ok = ActiveRecordUtils.with_transaction_and_rescue(results) do
-      mark_moab_not_found
-      complete_moab.save!
-    end
-    results.remove_db_updated_results unless transaction_ok
-    results.report_results
+  def update_complete_moab_status(status)
+    moab_validator.update_status(status)
+  end
+
+  # Moab and complete moab versions match?
+  def versions_match?
+    moab.current_version_id == complete_moab.version
+  end
+
+  def validate_versions
+    # set_status_as_seen_on_disk will update results and complete_moab
```

Comment:
method comments before the declaration, please.

Reply:
This is a comment about the next line, not about the method.